### PR TITLE
[SYCL] test errc for placeholder accessor fail

### PR DESCRIPTION
**Pull request by cperkinsintel**
update test to check for correct errc when attempting to use an unassociated placeholder accessor.

testing for PR https://github.com/intel/llvm/pull/5737

Signed-off-by: Chris Perkins <chris.perkins@intel.com>

### CODE REVIEW

File: SYCL/Basic/accessor/accessor.cpp [hunk 1]
```diff
@@ -556,6 +556,39 @@ int main() {
...
+          // we do NOT call .require(acc) without which we should throw a
+          // synchronous exception with errc::kernel_argument
+          cgh.parallel_for<class ph>(
+              r, [=](sycl::id<1> index) { acc[index] = 0; });
+        });
+        q.wait_and_throw();
+        assert(false && "we should not be here, missing exception");
+      } catch (sycl::exception &e) {
+        assert(e.code() == sycl::errc::kernel_argument &&
+               "incorrect error code");
```
Comment (steffenlarsen) [inline SYCL/Basic/accessor/accessor.cpp:585]: For debugging purposes it may be useful to have it print `e.what()` of unexpected SYCL exceptions.

Comment (cperkinsintel) [reply SYCL/Basic/accessor/accessor.cpp:585]: done

**Review by steffenlarsen — APPROVED** [review]

**Pull request merged**